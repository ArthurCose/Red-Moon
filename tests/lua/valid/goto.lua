print("shadowing and scope exit:")
do
  local x = 1
  local executed = false

  ::test::

  -- this should print 1 twice
  print(x)

  if executed then
    goto test_end
  end

  executed = true

  -- should be stored as a separate local, avoiding writing 2 into the other x variable
  local x = 2

  goto test
  ::test_end::
end

print("\nskip local:")
do
  do
    goto test

    local a = "a"
    print(a)

    ::test::
  end
  print("b")
end

print("\nskip scope:")
do
  goto test

  do
    local a = "a"
    print(a)
  end

  ::test::

  print("b")
end
