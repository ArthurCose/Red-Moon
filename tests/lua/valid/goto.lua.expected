shadowing and scope exit:
1
1

skip scope:
b
